samples/client/petstore/go/pet_api_test.go

```diff
@@ -58,7 +59,12 @@ func TestGetPetById(t *testing.T) {
 func TestGetPetByIdWithInvalidID(t *testing.T) {
 	resp, r, err := client.PetApi.GetPetById(context.Background(), 999999999)
```

Comment:
Are path parameters always their own argument? Can path parameters be optional? If so, how do we deal with this in the API?

Reply:
Currently all required parameters are a separate argument and all optional fields are in the optional struct, it does not differentiate path, body, qs, etc.

Reply:
So what happens if qs and body have the same parameters? Is the body seen as a type always or will it get split if we have multiple arguments? I'm a bit confused :D

Reply:
Same parameter will be put in both places. Interesting point, wonder what the other languages do.

Reply:
Have you seen my idea to have an embedded struct?

```go
type SomeRequestBody struct {
  Param1 NilString `json:"param1,omitempty"`
  Param2 string `json:"param2"`
  // ...
}
type SomeRequestPath struct {
  Param3 NilString `json:"param3,omitempty"`
  Param4 string `json:"param4"`
  // ...
}
type SomeRequestQuery struct {
  Param5 NilString `json:"param5,omitempty"`
  Param6 string `json:"param6"`
  // ...
}

type SomeRequest struct {
	*SomeRequestQuery
	*SomeRequestPath
	*SomeRequestBody 
}
```

That way it's possible to do:

```go
r := NewSomeRequest()
r.Param1 = ...
r.Param6 = ...
```

If there is a name collision, Go will print a compile error and you have to do:


```go
r := NewSomeRequest()
r.AType.Param1 = ...
r.BType.Param1 = ...
```

Reply:
This is possible, perhaps for the 3.0.0 version we can go this way.
Looking at the other 2.0.0 clients: PHP, Python, Perl, and Java take arguments the same way and do not differentiate their locations.

Reply:
Sounds good